hs_core/tests/api/native/test_copy_resource.py

```diff
@@ -0,0 +1,294 @@
+
+        new_res_raster = hydroshare.create_empty_resource(self.res_raster.short_id,
+                                                          self.nonowner,
+                                                          action='copy')
+        new_res_raster = hydroshare.copy_resource(self.res_raster, new_res_raster)
+
+        # test the new copied resource has the same resource type as the original resource
+        self.assertTrue(isinstance(new_res_raster, RasterResource))
+
+        # test science metadata elements are copied from the original resource to the new copied
+        # resource
+        self.assertTrue(new_res_raster.metadata.title.value == self.res_raster.metadata.title.value)
+        self.assertTrue(new_res_raster.creator == self.nonowner)
+
+        # test extended metadata elements are copied from the original resource to the new
+        # copied resource
+        if OriginalCoverage.objects.filter(object_id=self.res_raster.metadata.id).exists():
+            self.assertTrue(OriginalCoverage.objects.filter(
+                object_id=new_res_raster.metadata.id).exists())
+            self.assertEqual(new_res_raster.metadata.originalCoverage.value,
+                             self.res_raster.metadata.originalCoverage.value,
+                             msg="OriginalCoverage of new copied resource is not equal to "
+                                 "that of the original resource")
+
+        if CellInformation.objects.filter(object_id=self.res_raster.metadata.id).exists():
+            self.assertTrue(CellInformation.objects.filter(
+                object_id=new_res_raster.metadata.id).exists())
+            newcell = new_res_raster.metadata.cellInformation
+            oldcell = self.res_raster.metadata.cellInformation
+            self.assertEqual(newcell.rows, oldcell.rows,
+                             msg="Rows of new copied resource is not equal to that of "
+                                 "the original resource")
+            self.assertEqual(newcell.columns, oldcell.columns,
+                             msg="Columns of new copied resource is not equal to that of the "
+                                 "original resource")
+            self.assertEqual(newcell.cellSizeXValue, oldcell.cellSizeXValue,
+                             msg="CellSizeXValue of new copied resource is not equal to "
+                                 "that of the original resource")
+            self.assertEqual(newcell.cellSizeYValue, oldcell.cellSizeYValue,
+                             msg="CellSizeYValue of new copied resource is not equal to "
+                                 "that of the original resource")
+            self.assertEqual(newcell.cellDataType, oldcell.cellDataType,
+                             msg="CellDataType of new copied resource is not equal to "
+                                 "that of the original resource")
+
+        if BandInformation.objects.filter(object_id=self.res_raster.metadata.id).exists():
```

Comment:
The purpose for this check is to ensure the original raster resource to be copied has BandInformation object before checking this metadata element is copied to the target raster resource correctly given BandInformation is not a required field. Since the test raster resource does have this metadata element, I can remove this check as well as OriginalCoverage metadata element check.